### DISCUSSION TITLE
Broker instance naming (etc/broker.xml#/configuration/core/name)

### DISCUSSION THREAD

**Discussion by pekuz** [arkmq-org/arkmq-org-broker-operator#409] [Q&A] [answered]
Hi all,

we need to identify broker instances. The broker has in its core a notion of its own identity, configurable in _etc/broker.xml#/configuration/core/name_. Being configurable it allows to establish customized, to be meaningful, naming schemes. 

Now, I'm trying to understand if for the operator-controlled broker instances we could achieve customized naming scheme.

Reading https://artemiscloud.io/docs/help/operator/:

> Observe that the sample CR uses a naming convention of ex-aao. This naming convention denotes that the CR is an example resource for the ArtemisCloud (based on the ActiveMQ Artemis project) Operator. When you deploy this sample CR, the resulting Stateful Set uses the name ex-aao-ss. Furthermore, broker Pods in the deployment are directly based on the Stateful Set name, for example, ex-aao-ss-0, ex-aao-ss-1, and so on. The application name in the CR appears in the deployment as a label on the Stateful Set. You might use this label in a Pod selector, for example.

Kicking off with a concrete guess, the broker instance name could be as if _etc/broker.xml#/configuration/core/name_ element contained the Pod name (may be equals hostname too). So for the from-the-help example, the first broker instance would be named **ex-aao-ss-0** and the second broker instance would be named **ex-aao-ss-1**, and so on.

In other words 
 1. the broker instance name would be based on ActiveMQArtemis name, which is customizable, and 
 3. the broker instance name would allow to distinguish instances (0 vs 1) of the cluster.   

Would it make sense, please?

  Best regards
  Cc.

**Comment by brusdev** [comment, marked as answer]
This proposal makes sense to me, maybe we should add a new spec property in the ActiveMQArtemis CRD to avoid backward compatibility issues. Feel free to create a new feature issue at https://github.com/artemiscloud/activemq-artemis-operator/issues

**Comment by gtully** [comment]
Of note is that name attribute is used as the root of the JMX tree and the operator assumes that to be amq-broker, it is hard coded at the moment in the jolokia client used in the operator.
If the jolokia client was updated, then brokerProperties could be used set the name attribute and it could reference an env var to pull the identity out of the pod env, like host name or ordinal.
